### PR TITLE
Settings Screen

### PR DESCRIPTION
**Pull request by sravank9**
- Implemented Settings screen

https://user-images.githubusercontent.com/108011305/181016188-699f5c5c-968c-4e8c-9dc5-787783a925b1.mp4

### CODE REVIEW

File: app/src/main/java/com/dashxdemo/app/application/MyApplication.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.dashxdemo.app.application
+
+import android.app.Application
+import com.dashx.sdk.DashXClient
+
+class MyApplication: Application() {
+    override fun onCreate() {
+        super.onCreate()
+        DashXClient.createInstance(this,"TLy2w3kxf8ePXEyEjTepcPiq","https://api.dashx-staging.com/graphql","staging")
```
Comment (RabidFire) [inline app/src/main/java/com/dashxdemo/app/application/MyApplication.kt]: Keys shouldn't be hard-coded. We should provide best practices -- how the user would configure this (per environment) in their application.

File: app/src/main/java/com/dashxdemo/app/api/responses/SavedPreferencesResponse.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.dashxdemo.app.api.responses
+
+import com.google.gson.annotations.SerializedName
+
+data class SavedPreferencesResponse(
```
Comment (RabidFire) [inline app/src/main/java/com/dashxdemo/app/api/responses/SavedPreferencesResponse.kt]: Remove this class

File: app/src/main/java/com/dashxdemo/app/api/responses/StoredPreferenceResponse.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.dashxdemo.app.api.responses
+
+import com.google.gson.annotations.SerializedName
+
+data class StoredPreferenceResponse(
```
Comment (RabidFire) [inline app/src/main/java/com/dashxdemo/app/api/responses/StoredPreferenceResponse.kt]: Rename to StoredPreferencesResponse

File: app/build.gradle [hunk 4]
```diff
@@ -43,6 +43,9 @@ dependencies {
     implementation 'com.google.android.material:material:1.6.1'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
 
+    //DashX
```
Comment (RabidFire) [inline app/build.gradle]: Add space after `//`

File: app/src/main/java/com/dashxdemo/app/api/responses/StoredPreferenceResponse.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    @SerializedName("new-bookmark")
+    val newBookmark: StoredPreferenceData,
+    @SerializedName("new-post")
+    val newPost: StoredPreferenceData
+)
+
+data class StoredPreferenceData(
```
Comment (RabidFire) [inline app/src/main/java/com/dashxdemo/app/api/responses/StoredPreferenceResponse.kt]: We need to use the SDK class now

File: app/src/main/java/com/dashxdemo/app/application/MyApplication.kt (hunk 1 above)
Comment (RabidFire) [inline app/src/main/java/com/dashxdemo/app/application/MyApplication.kt]: These should not be hard-coded.

File: app/src/main/res/navigation/navigation_login.xml [hunk 6]
```diff
@@ -36,4 +36,5 @@
         android:id="@+id/nav_contact"
         android:name="com.dashxdemo.app.feature.login.fragment.ContactFragment"
         android:label="ContactFragment" />
+
```
Comment (RabidFire) [inline app/src/main/res/navigation/navigation_login.xml]: Remove extra space

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -32,6 +32,12 @@
     <string name="settings">Settings</string>
     <string name="update">Update</string>
 
+    <string name="whenCreateAPost">When someone creates a post</string>
+    <string name="createPostNotification">This will send a notification when someone creates a post</string>
+    <string name="whenBookmarksAPost">When someone bookmarks a post</string>
+    <string name="bookmarksPostNotification">"This will send a notification when someone bookmarks a post   "</string>
```
Comment (RabidFire) [inline app/src/main/res/values/strings.xml]: Why are there quotes and extra spaces here in this string?

Comment (RabidFire) [inline app/src/main/res/values/strings.xml]: The string names for the preferences can be based on the preference identifiers themselves:

```
- newPostNotification
- newPostNotificationHint
- newBookmarkNotification
- newBookmarkNotificationHint
```

**Review by RabidFire — CHANGES_REQUESTED** [review]

**Comment by sravank9** [comment]
@RabidFire  We need to push our SDK changes, to get this code run
We can get back here once the SDK changes were pushed to maven

File: app/build.gradle [hunk 8]
```diff
@@ -13,6 +13,13 @@ android {
         versionCode 1
         versionName '1.0.0'
 
+        Properties localProperties = new Properties()
+        localProperties.load(project.rootProject.file('local.properties').newDataInputStream())
+
+        buildConfigField 'String', 'DASHX_PUBLIC_KEY', localProperties.getProperty('DASHX_PUBLIC_KEY')
+        buildConfigField 'String', 'DASHX_BASE_URI', localProperties.getProperty('DASHX_BASE_URI')
+        buildConfigField 'String', 'DASHX_TARGET_ENVIRONMENT', localProperties.getProperty('DASHX_TARGET_ENVIRONMENT')
```
Comment (RabidFire) [inline app/build.gradle:21]: This seems like too much code. Maybe need to research some best practices around this @kiranattilicho

**Pull request merged**